core/src/main/java/org/infinispan/scattered/impl/ScatteredStateProviderImpl.java

```diff
@@ -75,34 +80,40 @@ public void start() {
             return CompletableFutures.completedNull();
          }
 
-         // we'll start at 1, so the counter will never drop to 0 until we send all chunks
+         // We pretend to do one extra invalidation at the beginning that finishes after we sent all chunks
          AtomicInteger outboundInvalidations = new AtomicInteger(1);
-         CompletableFuture<Void> outboundTaskFuture = new CompletableFuture<>();
-         OutboundTransferTask outboundTransferTask = new OutboundTransferTask(nextMember, oldSegments,
-            cacheTopology.getCurrentCH().getNumSegments(), chunkSize, cacheTopology.getTopologyId(), keyPartitioner,
-            task -> {
-               if (outboundInvalidations.decrementAndGet() == 0) {
-                  outboundTaskFuture.complete(null);
-               }
-            }, chunks -> invalidateChunks(chunks, otherMembers, outboundInvalidations, outboundTaskFuture, cacheTopology),
-            OutboundTransferTask::defaultMapEntryFromDataContainer, OutboundTransferTask::defaultMapEntryFromStore,
-            dataContainer, persistenceManager, rpcManager, commandsFactory, entryFactory, timeout, cacheName, true, true);
-         outboundTransferTask.execute(executorService);
-         return outboundTaskFuture;
+         CompletableFuture<Void> invalidationFuture = new CompletableFuture<>();
+         OutboundTransferTask outboundTransferTask =
+            new OutboundTransferTask(nextMember, oldSegments, cacheTopology.getCurrentCH().getNumSegments(), chunkSize,
+                                     cacheTopology.getTopologyId(), keyPartitioner,
+                                     chunks -> invalidateChunks(chunks, otherMembers, outboundInvalidations,
+                                                                invalidationFuture, cacheTopology),
+                                     OutboundTransferTask::defaultMapEntryFromDataContainer,
+                                     OutboundTransferTask::defaultMapEntryFromStore,
+                                     dataContainer, persistenceManager, rpcManager, commandsFactory, entryFactory,
+                                     timeout, cacheName, true, true);
+         outboundTransferTask.execute().whenComplete((ignored, throwable) -> {
```

Comment:
We probably want to log something if `throwable` is not null.

Reply:
I guess it isn't needed as we swallow all the exceptions further in, I have to admit it feels a little wrong though.

Reply:
I was debating whether I should do `thenAccept` instead when I wrote the code... I think I'll move the logging here instead.